Source/WebCore/html/HTMLButtonElement.cpp

```diff
@@ -140,7 +140,13 @@ void HTMLButtonElement::defaultEventHandler(Event& event)
     if (event.type() == eventNames.DOMActivateEvent && !isDisabledFormControl()) {
         RefPtr<HTMLFormElement> protectedForm(form());
 
-        if (protectedForm) {
+        if (commandForElement()) {
+            if (m_type != BUTTON && form())
+                return;
```

Comment:
I forgot what we decided here. This means that `<button type=submit commandfor=i>` no longer submits, even though there's no `command` attribute and even though `type=submit` can't be a command button anyway. Is that really what we want?

Reply:
For the `input` element it seems we're doing something completely different. Hmm.

Reply:
From my recollection we had decided that submit buttons with `commandfor` would no-op, to ward folks away from what was likely an author error.

Reply:
I guess it's reasonable if that happens even when `type` is not defaulted but set explicitly, as it's still an error.

I guess what I'm less sure on is whether both `command` and `commandfor` have to be set for that or if just `commandfor` suffices. As `<button type=button commandfor=i>` won't do anything either.

Reply:
Chrome logs some of these kinds of author errors to the console as warnings. I'm happy to do the same here but it doesn't seem like a convention in WebKit?

Reply:
I think we should do it where it seems useful. It might well be for cases like this.

Reply:
I'll follow up with this.